server/service/src/sync/test/pull_and_push.rs

```diff
@@ -74,14 +74,20 @@ async fn test_sync_pull_and_push() {
     let mut translated =
         translate_changelogs_to_push_records(&connection, changelogs.clone()).unwrap();
     translated.sort_by(|a, b| a.record.record_id.cmp(&b.record.record_id));
-    test_records.sort_by(|a, b| a.record_id.cmp(&b.record_id));
+
+    let mut test_records_to_push = test_records
+        .iter()
+        .filter(|r| r.is_sync_update == false)
```

Comment:
Do we have records in get_all_push_test_records that have is_sync_update == false ?

Btw this test fails to run, and I think for this test to fully test the `push` we would need to do different comparison rather than `replace_is_sync_update`:


```diff
+macro_rules! check_record_by_id_ignore_is_sync_update {
+    ($repository:ident, $connection:ident, $comparison_record:ident, $record_type:expr, $record_string:expr) => {{
+        let record = $repository::new(&$connection)
+        .find_one_by_id(&$comparison_record.id)
+        .unwrap()
+        .expect(&format!(
+            "{} row not found: {}",
+            $record_string, $comparison_record.id
+        ));
+        assert_eq!(
+            $record_type {
+                is_sync_update = false,
+                ..record
+            },
+            $record_type {
+                is_sync_update = false,
+                ..$comparison_record
+            }
+        )
+    }};
+}
+
 macro_rules! check_record_by_option_id {
     ($repository:ident, $connection:ident, $comparison_record:ident, $record_string:expr) => {{
         assert_eq!(
@@ -140,6 +162,7 @@ macro_rules! check_delete_record_by_id_option {
         )
     }};
 }
+
 pub(crate) async fn check_records_against_database(
     con: &StorageConnection,
     records: IntegrationRecords,
@@ -154,7 +177,13 @@ pub(crate) async fn check_records_against_database(
                 check_record_by_option_id!(StockLineRowRepository, con, record, "StockLine");
             }
             Name(record) => {
-                check_record_by_id!(NameRowRepository, con, record, "Name");
+                check_record_by_id_ignore_is_sync_update!(
+                    NameRowRepository,
+                    con,
+                    record,
+                    NameRow,
+                    "Name"
+                );
             }
             NameStoreJoin(record) => {
                 check_record_by_id!(NameStoreJoinRepository, con, record, "NameStoreJoin");
@@ -220,7 +249,13 @@ pub(crate) async fn check_records_against_database(
             }
 
             Clinician(record) => {
-                check_record_by_id!(ClinicianRowRepository, con, record, "Clinician")
+                check_record_by_id_ignore_is_sync_update!(
+                    ClinicianRowRepository,
+                    con,
+                    record,
+                    ClinicianRow,
+                    "Clinician"
+                )
             }
 
             ClinicianStoreJoin(record) => check_record_by_id!(
```